### ISSUE TITLE
Receipt returned by apple is different format

### ISSUE THREAD

**Issue by sambauwens** [Wizcorp/node-iap#10]
Hi!

While using your library I couldn't validate purchases, so I dug in the code and noticed that the receipt contained in the response from apple verification url isn't the one expected by the library.

-the bundle id is in receipt.bundle_id property instead of receipt.bid
-transaction_id and product_id are contained in receipt.in_app[0] instead of directly in the receipt.

I don't know why these values where located at different places for me, so I changed the code to test and made it work now.

I'll make a pull request to show you what I did and you can include it in the project if you want.

**Comment by sambauwens** [comment]
Also I noticed [another iap project](https://github.com/nov/itunes-receipt) using this in_app property.

**Comment by ronkorving** [comment]
Wow, that's really weird. But I appreciate the PR! :)

**Comment by nvcken** [comment]
I get response like this 
```
{ receipt: 
   { receipt_type: 'ProductionSandbox',
     adam_id: 0,
     app_item_id: 0,
     bundle_id: 'com.abc.test',
     application_version: '1.0',
     download_id: 0,
     version_external_identifier: 0,
     request_date: '2015-06-09 10:22:59 Etc/GMT',
     request_date_ms: '1433845379569',
     request_date_pst: '2015-06-09 03:22:59 America/Los_Angeles',
     original_purchase_date: '2013-08-01 07:00:00 Etc/GMT',
     original_purchase_date_ms: '1375340400000',
     original_purchase_date_pst: '2013-08-01 00:00:00 America/Los_Angeles',
     original_application_version: '1.0',
     in_app: [ [Object] ] },
  transactionId: undefined,
  productId: undefined,
  platform: 'apple' }
```
so weird,   transactionId: undefined,  productId: undefined
and have no status field like apple describe
https://developer.apple.com/library/ios/releasenotes/General/ValidateAppStoreReceipt/Chapters/ValidateRemotely.html#//apple_ref/doc/uid/TP40010573-CH104-SW1

What I miss?

**Comment by nvcken** [comment]
because the weird ( not like as document) format response
So this below code of module will work incorrect
lib/apple/index.js
```
/* jshint camelcase:false */
		if (payment.hasOwnProperty('productId') && payment.productId !== receipt.product_id) {
			return cb(new Error('Wrong product ID: ' + payment.productId + ' (expected: ' + receipt.product_id + ')'));
		}
```

**Comment by ronkorving** [comment]
That PR is still open, with unaddressed comments. An update to it would be most appreciated.

**Comment by nvcken** [comment]
322386b

**Comment by silenceper** [comment]
why the response is different?

**Comment by KLVTZ** [comment]
@nvcken @silenceper see my answer here:
https://github.com/Wizcorp/node-iap/pull/11#issuecomment-278722242

Also, check out #24 as this should resolve some of our issues --hopefully.

**Comment by KLVTZ** [comment]
@ronkorving I believe this can be closed due to the merge of #24

**Comment by ronkorving** [comment]
@sambauwens Please reopen if there's still an issue.
